### DISCUSSION TITLE
Ensure that addon hook is the last hook

### DISCUSSION THREAD

**Discussion by Ilke-dev** [mitmproxy/mitmproxy#5860] [Q&A]
Hi all,

I was wondering if it is possible to ensure that a specific addon can be set to be the last one be called on a specific hook. Would it be sufficient enough to set it as the last addon in the addon chain in the addonmanager?

**Comment by mhils** [comment]
> Would it be sufficient enough to set it as the last addon in the addon chain in the addonmanager?

Yes. By default, addon order is determined by https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/addons/__init__.py. Scripts provided by users are executed as part of `ScriptLoader`. You can of course manually change the order.

I could envision making this configurable at some point, for now we've managed to get surprisingly far by just picking a somewhat sensible order.

**Reply by Ilke-dev** [reply]
I was actually referring to this attribute: https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/addonmanager.py#L131
Because i want to be sure that my own addon is the last one to be called, so my train of thought was if i rearrange the order in the chain that i can ensure the hooks in my own addon are called last.

**Reply by mhils** [reply]
Yes, that works. :)
